File: clients/php7_client/client/core/base_strategy.php
<?php
abstract class BaseStrategy {
     /**
     * @var Debug
     */
    private $debug;

    function __construct($debug) {
        $this->debug = $debug;
    }

    /**
     * @param string $message
     */
    function log($message) {
        $this->debug->log($message);
    }

    function on_tick($my_passengers, $my_elevators, $enemy_passengers, $enemy_elevators) {
        return;
    }
}

File: clients/swift_client/client/BaseStrategy.swift
import Foundation

protocol Debug {
  func log(object:AnyObject)
  func `extension`(object:AnyObject)
}
protocol BaseStrategy {
  init(debug:Debug)
  func onTick(myElevators:[Elevator], enemyElevators:[Elevator], myPassenger:[Passenger], enemyPassenger: [Passenger])
}
